packages/documentation/src/stories/components/switch/switch.stories.tsx

```diff
@@ -0,0 +1,242 @@
+  },
+  argTypes: {
+    labelPosition: {
+      name: 'Label Position',
+      description: 'The position of the components label.',
+      control: {
+        type: 'radio',
+        label: {
+          both: 'Both',
+          before: 'Before',
+          after: 'After'
+        }
+      },
+      options: [
+        'both',
+        'before',
+        'after'
+      ],
+      table: {
+        category: 'General'
+      }
+    },
+    labelBefore: {
+      name: 'Label (before)',
+      description: 'Describes the content/topic of the component.',
+      if: {
+        arg: 'labelPosition',
+        neq: 'after'
+      },
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'General'
+      }
+    },
+    labelAfter: {
+      name: 'Label (after)',
+      description: 'Describes the content/topic of the component.',
+      if: {
+        arg: 'labelPosition',
+        neq: 'before'
+      },
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'General'
+      }
+    },
+    hiddenLabel: {
+      name: 'Hidden Label',
+      description: '<p>Render the component with or without a visible label.</p><div className="alert alert-info alert-sm">There are accessibility issues with hidden labels.<br/>Please read our <a href="/?path=/story/foundations-accessibility--page#labels">labels accessibility guide</a>.</div>',
+      control: {
+        type: 'boolean'
+      },
+      table: {
+        category: 'General'
+      }
+    },
+    checked: {
+      name: 'Checked',
+      description: 'Defienes the checked state of the component.',
+      control: {
+        type: 'boolean'
+      },
+      table: {
+        category: 'States'
+      }
+    },
+    disabled: {
+      name: 'Disabled',
+      description: '<p>When set to `true`, disables the component\'s functionality and places it in a disabled state.</p><div className="alert alert-info alert-sm">There are accessibility issues with the disabled state.<br/>Please read our <a href="/?path=/docs/foundations-accessibility--page#disabled-state">disabled state accessibility guide</a>.</div>',
+      control: {
+        type: 'boolean'
+      },
+      table: {
+        category: 'States'
+      }
+    },
+    validation: {
+      name: 'Validation',
+      description: 'Controls the validation state appearance of the component.',
+      control: {
+        type: 'radio',
+        labels: {
+          'null': 'Default',
+          'is-valid': 'Valid',
+          'is-invalid': 'Invalid'
+        }
+      },
+      options: [
+        'null',
+        'is-valid',
+        'is-invalid'
+      ],
+      table: {
+        category: 'States'
+      }
+    },
+    validFeedback: {
+      name: 'Valid Feedback',
+      description: 'Text to show when the component is in an valid state.',
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'States'
+      }
+    },
+    invalidFeedback: {
```

Comment:
Valid and invalid feedback controls can be displayed conditionally based on the selected validation (using `if: { arg: '...', eq: '...'}`)

Reply:
I removed them instead, because the user has to dynamically set them in most cases anyway.